### PR TITLE
Fix child process spawn on macos hangs with stdin, stdout behavior set to pipe

### PR DESCRIPTION
**Pull request by rabingaire**
This PR fixes the issue mentioned in issue #11470

### CODE REVIEW

File: lib/std/child_process.zig [hunk 1]
```diff
@@ -1375,3 +1379,49 @@ test "build and call child_process" {
     const ret_val = try child_proc.spawnAndWait();
     try testing.expectEqual(ret_val, .{ .Exited = 0 });
 }
+
+test "creating a child process with stdin/stdout behavior set to StdIo.Pipe" {
```
Comment (matu3ba) [inline lib/std/child_process.zig:1383]: nit: please dont write `stdin/stdout` as it is annoying to filter the test in shell on error

File: lib/std/child_process.zig [hunk 2]
```diff
@@ -1375,3 +1379,49 @@ test "build and call child_process" {
...
+                \\const std = @import("std");
+                \\pub fn main() void {
+                \\    std.debug.print("Hello World", .{});
+                \\}
+                \\
+            ;
+            try testing.expectEqualStrings(expected_program, out_bytes);
+        },
+        else => {
+            try testing.expect(false);
```
Comment (matu3ba) [inline lib/std/child_process.zig:1424]: nit: Typically, `unreachable` is used instead.

**Review by matu3ba — COMMENTED** [review]
change is looking nice and clean.

only 2 nits.

**Comment by rabingaire** [comment]
@matu3ba thank you for the comments will make the changes

**Review by kubkon — APPROVED** [review]
Looks great, thanks @rabingaire, and thanks @matu3ba for a review!

**Pull request merged**